src/containers/SecretsModal/SecretsModal.js

```diff
@@ -23,21 +23,32 @@ import { getServiceAccounts, isWebSocketConnected } from '../../reducers';
 import { fetchServiceAccounts } from '../../actions/serviceAccounts';
 
 /* istanbul ignore next */
-function validateInputs(value, id) {
+function validateInputs(value, id, namespace) {
   const trimmed = value.trim();
 
   if (trimmed === '') {
     return false;
   }
   if (id === 'name' || id === 'serviceAccount') {
-    if (trimmed.length > 253) {
+    if (trimmed.length >= 253) {
       return false;
     }
 
-    if (/[^-.a-z0-9]/.test(trimmed)) {
+    if (/[^-a-z0-9]/.test(trimmed)) {
       return false;
     }
   }
+  if (id === 'name') {
+    if (trimmed.startsWith('-', 0) || trimmed.endsWith('-')) {
+      return false;
+    }
+    if (namespace === undefined) {
+      return false;
+    }
+    if (namespace === 'tekton-pipelines') {
```

Comment:
this isn't needed as it'll just fall through to the `return true` below anyway

Reply:
My bad, i left that in from testing something worked when helping Megan
Removing it